### PR TITLE
Backports v0.18 PR's: 14805, 14947 and 14953

### PR DESCRIPTION
**Pull request by pravblockc**
Backports v0.18 PR's: 14805, 14947 and 14953

### CODE REVIEW

File: src/test/denialofservice_tests.cpp [hunk 1]
```diff
@@ -134,7 +134,7 @@ BOOST_AUTO_TEST_CASE(outbound_slow_chain_eviction)
 static void AddRandomOutboundPeer(std::vector<CNode *> &vNodes, PeerLogicValidation &peerLogic, CConnmanTest* connman)
 {
     CAddress addr(ip(g_insecure_rand_ctx.randbits(32)), NODE_NONE);
-    vNodes.emplace_back(new CNode(id++, ServiceFlags(NODE_NETWORK|NODE_WITNESS), 0, INVALID_SOCKET, addr, 0, 0, CAddress(), "", /*fInboundIn=*/ false));
+    vNodes.emplace_back(new CNode(id++, ServiceFlags(NODE_NETWORK), 0, INVALID_SOCKET, addr, 0, 0, CAddress(), "", /*fInboundIn=*/ false));
```
Comment (PastaPastaPasta) [inline src/test/denialofservice_tests.cpp]: this part of e6d9fadc0a9c4d3baef1ae69b9ebd4f5b5e1f9d0 squash into 14953 merge

File: test/functional/test_framework/test_framework.py [hunk 2]
```diff
@@ -321,7 +321,7 @@ def add_nodes(self, num_nodes, extra_args=None, *, rpchost=None, binary=None):
         assert_equal(len(binary), num_nodes)
         for i in range(num_nodes):
             numnode = len(self.nodes)
-            self.nodes.append(TestNode(numnode, get_datadir_path(self.options.tmpdir, numnode), rpchost=rpchost, timewait=self.rpc_timewait, bitcoind=binary[i], bitcoin_cli=self.options.bitcoincli, mocktime=self.mocktime, coverage_dir=self.options.coveragedir, extra_conf=extra_confs[i], extra_args=extra_args[i], use_cli=self.options.usecli))
+            self.nodes.append(TestNode(numnode, get_datadir_path(self.options.tmpdir, numnode), self.extra_args_from_options, chain=self.chain, rpchost=rpchost, timewait=self.rpc_timewait, bitcoind=binary[i], bitcoin_cli=self.options.bitcoincli, mocktime=self.mocktime, coverage_dir=self.options.coveragedir, extra_conf=extra_confs[i], extra_args=extra_args[i], use_cli=self.options.usecli))
```
Comment (PastaPastaPasta) [inline test/functional/test_framework/test_framework.py]: this part of e6d9fad squash into 14805 merge

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
Looks very good! 

Just please squash those changes into the original PR, and it looks good

File: src/test/denialofservice_tests.cpp (hunk 1 above)
Comment (pravblockc) [reply src/test/denialofservice_tests.cpp]: Resolved

File: test/functional/test_framework/test_framework.py (hunk 2 above)
Comment (pravblockc) [reply test/functional/test_framework/test_framework.py]: Fixed

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**